docs/design/proposals/adoption/resource-adoption.md

```diff
@@ -0,0 +1,225 @@
+# Resource Adoption 
+
+# The Problem
+
+New ACK users typically already have implemented their infrastructure in existing solutions, meaning they have deployed production AWS resources into their accounts. While ACK provides good tooling for creating new resources, the path to migration from existing declarative infrastructure technologies into ACK is, so far, bumpy. The current method for migration requires users to store the current state of their applications, delete the resources, and re-create them using ACK. This process introduces numerous places for bugs, loss of the state of existing deployments and downtime in production workloads. Furthermore, considering the case where a K8s cluster loses state, users have no path to recover the previous resources without having to recreate all of them using ACK.
```

Comment:
> we cannot use the same CRD as the Kind itself because the fields in the Spec that describe the desired state of the Kind are typically marked as required

I would say this is a potential problem with other of the possible solutions, but it's not a problem that exists in ACK today.

Reply:
> > we cannot use the same CRD as the Kind itself because the fields in the Spec that describe the desired state of the Kind are typically marked as required
> 
> I would say this is a potential problem with other of the possible solutions, but it's not a problem that exists in ACK today.

Sure. I say "exists today" just because the `services.k8s.aws/arn` "solution" exists today -- but use of that annotation is problematic for the reasons described.

Reply:
> you might consider listing a set of Use Cases in a separate section, one of which might mention the use case of an existing Terraform or CloudFormation user where TF/CF has been responsible for managing the lifecycle of an existing S3 Bucket and the user is interested in describing that S3 Bucket resource as a Kubernetes CRD and having ACK manage the resource's lifecycle.

+1 to this. I'd make it clear very early on that resource adoption will just "adopt" the AWS resources, not making any YAML/HCL -> CRD translation or automatic conversion.

Reply:
> not making any YAML/HCL -> CRD translation or automatic conversion.

By this do you mean Cfn YAML -> CRD? Because this design does suggest we will be able to generate an ACK CRD for each resource, as part of the adoption.

Reply:
Yup, when I said YAML I was thinking of CloudFormation YAML. I should've been more specific 🤦‍♂️